federation/pkg/kubefed/unjoin.go

```diff
@@ -81,6 +81,16 @@ func unjoinFederation(f cmdutil.Factory, cmdOut, cmdErr io.Writer, config util.A
 		return nil
 	}
 
+	//we need to ensure deleting the config map created in the deregistered cluster
+	//this configmap was created when the cluster joined this federation to aid
+	//the kube-dns of that cluser to aid service discovery
+	unjoinClusterFactory := config.HostFactory(unjoinFlags.Name, unjoinFlags.Kubeconfig)
```

Comment:
I am not entirely sure if this works. What if the cluster name in `unjoinFlags.Name` and the context name in `kubeconfig` are not the same?